app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue

```diff
@@ -114,6 +114,52 @@ const selfArgumentExternalId = computed<Opt<ExternalId>>(() => {
 const visualizationConfig = computed<Opt<NodeVisualizationConfiguration>>(() => {
   // Even if we inherit dynamic config in props.input.dynamicConfig, we should also read it for
   // the current call and then merge them.
+  function collectArgumentNamesAndUuids(v : Ast.Ast) {
+    var arr : Array<any> = [];
+    v.visitRecursiveAst(function(f) {
```

Comment:
`visitRecursive` is too deep; it will pick up names from subexpressions. Only the current app chain should be considered. Something like:
```ts
let arr = []
let ast = v
while (ast instanceof Ast.App) {
    arr.push(... info ...)
    ast = ast.function
}
```

Reply:
Also, in this widget we already have a concept of "interpreted application" (see `interpreted` at line ~48) you have all arguments with their names already there.

Reply:
Thanks @farmaazon, I am trying to use it, but:
```
let args = ArgumentApplication.FromInterpretedWithInfo(interpreted.value)
console.log(args)
```
and now what? I am totally lost when I see

![image](https://github.com/enso-org/enso/assets/26887752/52b0d92d-7aea-491f-831b-fa3aae2334e9)

Reply:
Looks good. `args.argument.ast` should be exactly the same object you're handling as `f` in current `collectArgumentNamesAndUuids` implementation. `args.argument.ast.code()` should give you nice code representation.

To iterate entire app chain, you may use `args.iterApplications`, and check `.argument.ast` of every received application.

Reply:
Thank you for your guidance, Adam. Done in a6fb11d